tck/src/main/java/org/eclipse/configjsr/DynamicConfigSourceTest.java

```diff
@@ -65,30 +67,38 @@ public static WebArchive deploy() {
     @Test
     public void testBgCount() throws Exception {
         Integer value = config.getValue(DynamicChangeConfigSource.TEST_ATTRIBUTE, Integer.class);
-        Thread.sleep(20L);
+        Thread.sleep(25L);
         Integer value2 = config.getValue(DynamicChangeConfigSource.TEST_ATTRIBUTE, Integer.class);
         assertTrue(value2 > value);
     }
 
     @Test
-    public void testBgCallback() throws Exception {
-        Integer value = config.getValue(DynamicChangeConfigSource.TEST_ATTRIBUTE, Integer.class);
-        Map<String, Integer> vals = new ConcurrentHashMap<>();
-
-
-        config.registerConfigChangedListener(s -> s.forEach(k -> vals.put(k, config.getValue(k, Integer.class))));
-        vals.clear();
-
-        Thread.sleep(12L);
-        assertEquals(1, vals.size());
-
-        int i1 = vals.get(DynamicChangeConfigSource.TEST_ATTRIBUTE);
-
-        Thread.sleep(15L);
-        assertEquals(1, vals.size());
-
-        int i2 = vals.get(DynamicChangeConfigSource.TEST_ATTRIBUTE);
-        assertTrue(i2 > i1);
+    public void testValueInvalidationOnConfigChange() throws Exception {
+        ConfigValue<Integer> valCfg = config.access(DynamicChangeConfigSource.TEST_ATTRIBUTE)
```

Comment:
I considered this to be optional approach by the implementor. This behavior should not be tested - the implementation may decide not to refresh the value until the cache for is expired.

Reply:
Well, the intention is to define that it *must* expire the cache _if_ the Config gets notified about a change by the ConfigSource. Do you really think we should not mandate this?

Reply:
We need to find a way to test this. Otherwise, how to mandate implementor implement this?

Reply:
There should be a test for it already. It sets a cacheFor to 15 minutes, then loads the values. Then the ConfigSource changes a value via the TCK test and fires the attributeChanged to the Config. And then the test reads the value again and verifies that the new value is picked up immediately.